### PR TITLE
Deprecated Person model and endpoints

### LINKED ISSUE #24: Deprecate Person object

### LINKED ISSUE #25: Restucture Person to User
Deprecate Person endpoints and implement new user endpoints and user model

### PR DESCRIPTION
**Pull request by Iiro127**
Resolves #24 
Resolves #25

### CODE REVIEW

File: swagger.yaml [hunk 1]
```diff
@@ -744,9 +804,46 @@ components:
...
+        email:
+          type: string
+          description: Person email.
+        isActive:
+          type: boolean
+          description: Defining is person active e.g. is employed.
+        severaGuid:
+          type: integer
+          format: int32
+          description: Persons GUID in Severa
```
Comment (villejuutila) [inline swagger.yaml]: User GUID in Severa seems to be of type String [https://api.severa.visma.com/rest-api/doc/index.html#/UsersRead/Users_GetUsers](https://api.severa.visma.com/rest-api/doc/index.html#/UsersRead/Users_GetUsers)

File: swagger.yaml [hunk 2]
```diff
@@ -23,11 +23,70 @@ paths:
...
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+
+  /v1/users/{guid}:
+    get:
+      operationId: findUser
+      summary: Find user by id
+      description: Finds user by id in Severa.
```
Comment (villejuutila) [inline swagger.yaml]: Aren't you supposed to find users by their Keycloak ID?

**Review by villejuutila — CHANGES_REQUESTED** [review]

File: swagger.yaml (hunk 1 above)
Comment (Iiro127) [reply swagger.yaml]: Resolved

File: swagger.yaml (hunk 2 above)
Comment (Iiro127) [reply swagger.yaml]: Resolved

File: swagger.yaml [hunk 3]
```diff
@@ -23,11 +23,69 @@ paths:
               schema:
                 type: string
 
+  /v1/users:
+    get:
+      operationId: listUsers
+      summary: Get list of Severa users.
+      description: Get list of Severa users.
```
Comment (villejuutila) [inline swagger.yaml]: Are these Severa users?

File: swagger.yaml [hunk 4]
```diff
@@ -23,11 +23,69 @@ paths:
...
+        - Users
+      parameters:
+        - name: isActive
+          in: query
+          description: Filter by users active state
+          schema:
+            type: boolean
+      responses:
+        "200":
+          description: Get list of Severa users
```
Comment (villejuutila) [inline swagger.yaml]: Are these Severa users?

File: swagger.yaml [hunk 5]
```diff
@@ -23,11 +23,69 @@ paths:
...
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+
+  /v1/users/{uuid}:
+    get:
+      operationId: findUser
+      summary: Find user by keycloak id
+      description: Finds user by keycloak id in keycloak.
```
Comment (villejuutila) [inline swagger.yaml]: Parameter named uuid with type UUID is a bit ambigious. Typically REST-endpoints are named /{version}/{resource in plural}/{identifier of resource}, with identifier being the Users id in this case. Or explicitly `{resource in singular}Id`, I'd call this `/v1/users/{userId}`.

File: swagger.yaml [hunk 6]
```diff
@@ -23,11 +23,69 @@ paths:
...
+      tags:
+        - Users
+      parameters:
+        - name: uuid
+          in: path
+          required: true
+          description: keycloak uuid
+          schema:
+            type: string
```
Comment (villejuutila) [inline swagger.yaml]: Parameter named uuid with type UUID is a bit ambigious. Typically REST-endpoints are named /{version}/{resource in plural}/{identifier of resource}, with identifier being the Users id in this case. Or explicitly `{resource in singular}Id`, I'd call this `/v1/users/{userId}`.

```      
       parameters:
         - name: userId
           in: path
           required: true
           description: Users ID in Keycloak
           schema:
             type: string
             format: uuid
```

File: swagger.yaml [hunk 7]
```diff
@@ -54,9 +112,10 @@ paths:
 
   /v1/persons/{personId}:
     get:
+      deprecated: true
       operationId: findPerson
       summary: Find person by id.
-      description: Finds person by id in Keycloak.
+      description: Finds person by id in Keycloak (Deprecated, use /v1/users/{guid}).
```
Comment (villejuutila) [inline swagger.yaml]:
```suggestion
      description: Finds person by id in Keycloak (Deprecated, use /v1/users/{userId}).
```

**Review by villejuutila — CHANGES_REQUESTED** [review]

File: swagger.yaml (hunk 3 above)
Comment (Iiro127) [reply swagger.yaml]: Resolved

File: swagger.yaml (hunk 4 above)
Comment (Iiro127) [reply swagger.yaml]: Resolved

File: swagger.yaml (hunk 6 above)
Comment (Iiro127) [reply swagger.yaml]: Resolved

File: swagger.yaml (hunk 5 above)
Comment (Iiro127) [reply swagger.yaml]: Resolved

File: swagger.yaml (hunk 7 above)
Comment (Iiro127) [reply swagger.yaml]: Resolved

**Review by villejuutila — APPROVED** [review]

**Pull request merged**